### DISCUSSION TITLE
Help with merge throwing an error

### DISCUSSION THREAD

**Discussion by Emil-cmd** [mtex-toolbox/mtex#1441] [Ask Anything]
Hello 

I am right now trying to merge a subset of grains (in this case recrystallised) that contain a CSL3 boundary. The material in question is a Ni-base superalloy. The maps are taken from samples deformed to different strains ranging from 0.1 to 0.8
My code looks as follows: 

[mergedGrains,parentIds] = merge(grains_ReX,CSL); 

This only works for the samples deformed to the highest strains, 0.4, 0.6 and 0.8. For the three lowest, 0.1, 0.15 and 0.2 Matlab gives me the following error 

Error using sparse
Index exceeds array bounds.

Error in grain2d/merge (line 68)
    A = A | sparse(mergeId(:,1),mergeId(:,2),1,maxId+1,maxId+1);

I cannot figure out why the same line of code only works for some of the samples and not for the others. I have tried using the CSL-boundaries variable as only be the CSL boundaries in the specific ReX-grains (CSL_ReX) but also for all the present grains (CSL). The result is the same where the code works and gives the same result for the three higher strains and throws me this error for the lower. 

I do not know if it can be because there are quite few ReX grains for the lower strains and the subset is therefore quite small, but I can't see how that should matter. I have tried doing the merge using all the grains, instead of just a subset of recrystallised grains and then it works. However, I cannot proceed with my analysis with that information. I am only interested in merging recrystallised grains with a CSL3 boundary between them. 

The data for one of the samples that yields the error can be found here
https://1drv.ms/u/s!Amq0FZ71eAbmiI5IQwSUSf08OnuoJg?e=BINpnJ

Can anyone help me figure out what causes this error? Thank you in advance.

**Comment by kilir** [comment]
Hi,
In those samples/subsets that fail, do you freshly compute the grains of only that subset? It looks like this might not be the case.
Alternatively, merging first and separating rxx/non-rxx afterwards might be a solution.

Cheers,
Rüdiger

**Reply by Emil-cmd** [reply]
Hello Rüdiger.
No, the subset is acquired from the complete subset. Basically the ReX grains are grains that fulfil the critieria grain.GOS < critical_GOS_value. 
Do you think the error might resolve if I extract the ebsd-data from the ReX grains and redo the calcGrains procedure on them independently? 

Merging the complete set of grains would maybe also work but for this specific case be very time consuming since there are a lot of MC-carbides that need to be manually removed. This has been done, but merging the entire set would give these carbides new Id's and the process of manually remove them would need to be redone. I am hoping to avoid this. 
A question that arises however is what would happen with the GOS value of the merged gains? Removing a CSL3 boundary would introduce a 60 deg rotation in the merged grain, which would in turn create large GOS-values. I could be that it would be difficult to separate the ReX and def. grains based on GOS with this approach as well.

**Reply by kilir** [reply]
I see that that the GOS won't work anymore with the merged grains.
In that case, I'd say either reconstructing grains from the ebsd subset might be a solution.- Alternatively, you can use the id the IDs of the pre-merged rxx grains to find them later in the full set of merged grains.
Rüdiger
